src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.cpp

```diff
@@ -0,0 +1,63 @@
+#include "ai/hl/stp/tactic/block_shot_path_tactic.h"
+
+#include "geom/util.h"
+#include "shared/constants.h"
+#include "util/logger/init.h"
+
+BlockShotPathTactic::BlockShotPathTactic(bool loop_forever) : Tactic(loop_forever) {}
+
+std::string BlockShotPathTactic::getName() const
+{
+    return "Block Shot Path Tactic";
+}
+
+void BlockShotPathTactic::updateParams(const Robot& enemy_robot, const Field& field)
+{
+    updateParams(enemy_robot.position(), field);
+}
+
+void BlockShotPathTactic::updateParams(const Point& shot_origin, const Field& field)
+{
+    // Update the parameters stored by this Tactic
+    this->shot_origin = shot_origin;
+    this->field       = field;
+}
+
+double BlockShotPathTactic::calculateRobotCost(const Robot& robot, const World& world)
+{
+    // Prefer robots closer to the block position
+    // We normalize with the total field length so that robots that are within the field
+    // have a cost less than 1
+    Point block_position = getBlockPosition();
+    double cost = (robot.position() - block_position).len() / world.field().totalLength();
+    return std::clamp<double>(cost, 0, 1);
+}
+
+Point BlockShotPathTactic::getBlockPosition()
+{
+    if (!this->field)
+    {
+        LOG(WARNING) << "The Field for the " << getName() << " was not initialized"
+                     << std::endl;
+        return this->robot->position();
+    }
```

Comment:
Could we have the field be passed into the constructor? Then we could remove this check.

Reply:
Yea that seems reasonable. _Technically_ this is more robust to the field changing, but IRL it should never change while AI is running.

Reply:
fixed